src/main/webapp/view/transcript/transcript.js

```diff
@@ -112,7 +112,13 @@ function deleteTranscript() {
 /** Seeks transcript to {@code currentTime}, which is given in seconds. */
 function seekTranscript(currentTime) {
   const currentTimeMs = window.secondsToMilliseconds(currentTime);
-  if (currentTimeMs <= currentTranscriptLine.endTimestampMs) {
+  if (currentTimeMs < currentTranscriptLine.startTimestampMs) {
+    return;
+  }
+  if (isWithinCurrentTimeRange(currentTimeMs)) {
+    if (!isBolded(currentTranscriptLine)) {
+      addBold(currentTranscriptLine);
```

Comment:
This addBold method could check if the line is already bold and return early if it is. The benefit is that you do not need to have nested if-statements.

Reply:
done! I also added a similar check to `removeBold`.